openwisp_monitoring/monitoring/base/models.py

```diff
@@ -39,12 +40,13 @@
 
 
 class AbstractMetric(TimeStampedEditableModel):
+    METRICS = get_metric_configuration()
     name = models.CharField(max_length=64)
-    description = models.TextField(blank=True)
-    key = models.SlugField(
-        max_length=64, blank=True, help_text=_('leave blank to determine automatically')
-    )
+    key = models.SlugField(max_length=64, blank=True)
     field_name = models.CharField(max_length=16, default='value')
+    configuration = models.CharField(
+        max_length=16, null=True, choices=get_metric_configuration_choices()
```

Comment:
Not really sure about the chocies parameter. Everytime you add a new metric configuration will make Django generate a new migration. @nemesisdesign WDYT?

Reply:
My concern is solved using the `register_chart` function :+1: